### DISCUSSION TITLE
Save more data in properties file

### DISCUSSION THREAD

**Discussion by MikeSchumacherCapgemini** [devonfw/cobigen#1416] [Ideas]
As a Cobigen user, I want Cobigen to save recurring data in a properties file so that i do not have to repeatedly take care of it.

For example, if Cobigen can recognize that Spring Boot is used in a project, it can automatically adapt to it by adding relevant annotations, among others. The same would be true for similar project-defining dependencies.
This also enables Cobigen to not show increments that are incompatible/not useful with the frameworks that are used.
Cobigen could also add data to the properties file by asking the user or saving the users input (If the user uses templates for one thing, don't show the other ones)(Should be configurable)

This is a workshop topic, any ideas are welcome and can be discussed in the comments

**Comment by hohwille** [comment]
Yes, we had "DAO vs. Repository" or "Complex vs. simple logic (with or without UCs)" and alike increments where we simply offered all options to end-users.
However, regular developers are confused with such decisions that actually have already been made by the architect so adding something to a `cobigen.properties` file to "persist" this decision and then have properties enabling or disabling increments, templates, template-elements (methods, fields, constructors) via such properties would be a very nice enhancement.
